jaxrs/src/main/java/org/killbill/billing/jaxrs/json/CatalogJson.java

```diff
@@ -0,0 +1,549 @@
+            }
+
+            return true;
+        }
+
+        @Override
+        public int hashCode() {
+            int result = type != null ? type.hashCode() : 0;
+            result = 31 * result + (name != null ? name.hashCode() : 0);
+            result = 31 * result + (plans != null ? plans.hashCode() : 0);
+            result = 31 * result + (included != null ? included.hashCode() : 0);
+            result = 31 * result + (available != null ? available.hashCode() : 0);
+            return result;
+        }
+    }
+
+    public static class PlanJson {
+
+        private final String name;
+        private final BillingPeriod billingPeriod;
+        private final List<PhaseJson> phases;
+
+        @JsonCreator
+        public PlanJson(@JsonProperty("name") final String name,
+                        @JsonProperty("billingPeriod") final BillingPeriod billingPeriod,
+                        @JsonProperty("phases") final List<PhaseJson> phases) {
+            this.name = name;
+            this.billingPeriod = billingPeriod;
+            this.phases = phases;
+        }
+
+        public String getName() {
+            return name;
+        }
+
+        public BillingPeriod getBillingPeriod() {
+            return billingPeriod;
+        }
+
+        public List<PhaseJson> getPhases() {
+            return phases;
+        }
+
+        @Override
+        public String toString() {
+            final StringBuilder sb = new StringBuilder("PlanJson{");
+            sb.append("name='").append(name).append('\'');
+            sb.append("billingPeriod='").append(billingPeriod).append('\'');
+            sb.append(", phases=").append(phases);
+            sb.append('}');
+            return sb.toString();
+        }
+
+        @Override
+        public boolean equals(final Object o) {
+            if (this == o) {
+                return true;
+            }
+            if (o == null || getClass() != o.getClass()) {
+                return false;
+            }
+
+            final PlanJson planJson = (PlanJson) o;
+
+            if (name != null ? !name.equals(planJson.name) : planJson.name != null) {
+                return false;
+            }
+            if (billingPeriod != null ? !billingPeriod.equals(planJson.billingPeriod) : planJson.billingPeriod != null) {
+                return false;
+            }
+            if (phases != null ? !phases.equals(planJson.phases) : planJson.phases != null) {
+                return false;
+            }
+
+            return true;
+        }
+
+        @Override
+        public int hashCode() {
+            int result = name != null ? name.hashCode() : 0;
+            result = 31 * result + (billingPeriod != null ? billingPeriod.hashCode() : 0);
+            result = 31 * result + (phases != null ? phases.hashCode() : 0);
+            return result;
+        }
+    }
+
+    public static class PhaseJson {
+
+        private final String type;
+        private final List<PriceJson> prices;
```

Comment:
@maguero I merged the PR as it was, but it would probably make a lot of sense to add those fields in the `PhaseJson` because otherwise the client of that api would miss critical info about the catalog. Is that something you could add in a separate PR?